App.tsx

```diff
@@ -11,51 +12,87 @@ import Registration from 'screens/registration/Registration';
 import DrawerNavigator from 'components/navigators/DrawerNavigator';
 import HotSpots from 'screens/hotSpots/HotSpots';
 import SuperFood from 'screens/superFood/SuperFood';
+import {initDB} from 'services/DBService';
+import Home from 'screens/home/Home';
+import {AppDispatch, RootState, store} from 'store/store';
+import {setLoginStatus} from 'store/slice/userSlice';
+import MealDetail from 'screens/mealDetail/MealDetail';
 
-const App = () => {
+const AppContent = () => {
   const Stack = createNativeStackNavigator<RootStackParamList>();
+  const dispatch = useDispatch<AppDispatch>();
+  const isLoggedIn = useSelector((state: RootState) => state.user.isLoggedIn);
 
-  const [isLoggedIn, setIsLoggedIn] = useState<boolean | null>(null);
+  const [hasCompletedOnboarding, setHasCompletedOnboarding] = useState(false);
 
   useEffect(() => {
-    const checkLoginStatus = async () => {
+    const checkInitialStates = async () => {
       try {
-        const token = await AsyncStorage.getItem('uid');
-        setIsLoggedIn(!!token);
+        const onboardingStatus = await AsyncStorage.getItem(
+          'hasCompletedOnboarding',
+        );
+        setHasCompletedOnboarding(onboardingStatus === 'true');
+
+        const uid = await AsyncStorage.getItem('uid');
+        dispatch(setLoginStatus(!!uid));
       } catch (error) {
-        setIsLoggedIn(false);
+        dispatch(setLoginStatus(false));
       } finally {
         SplashScreen.hide();
       }
     };
 
-    checkLoginStatus();
-  }, []);
+    const initializeDatabase = async () => {
+      await initDB();
+    };
 
-  if (isLoggedIn === null) {
-    // Optional: Show a loader while checking login status
-    return null;
+    initializeDatabase();
+    checkInitialStates();
+  }, [dispatch]);
+  if (isLoggedIn === null || hasCompletedOnboarding === null) {
```

Comment:
you should use isNil method of lodash at here like below
```suggestion
  if (isNil(isLoggedIn) || isNil(hasCompletedOnboarding)) {
```

Reply:
fixed